### PR TITLE
Align buffer size for Azure and AWS functions

### PR DESCRIPTION
**Pull request by loicmathieu**
Some small implementation improvements for Azure function HTTP and Amazon lambda HTTP.
This align the implementation with Google Cloud Functin HTTP.

- Use a `ByteArrayOutputStream` of 8k as it's the admitted default ideal size (this is the size used inside the JDK by default)
- Close the `ByteArrayOutputStream` gracefully

### CODE REVIEW

File: extensions/amazon-lambda-http/runtime/src/main/java/io/quarkus/amazon/lambda/http/LambdaHttpHandler.java [hunk 1]
```diff
@@ -38,7 +39,9 @@
 
 @SuppressWarnings("unused")
 public class LambdaHttpHandler implements RequestHandler<AwsProxyRequest, AwsProxyResponse> {
-    private static final Logger log = Logger.getLogger("quarkus.amazon.lambda.http");
+    private static final Logger LOG = Logger.getLogger("quarkus.amazon.lambda.http");
```
Comment (patriot1burke) [inline extensions/amazon-lambda-http/runtime/src/main/java/io/quarkus/amazon/lambda/http/LambdaHttpHandler.java:42]: Please avoid doing subjective changes like this  (changing variable names) in the future.

File: extensions/azure-functions-http/runtime/src/main/java/io/quarkus/azure/functions/resteasy/runtime/BaseFunction.java [hunk 2]
```diff
@@ -156,6 +158,11 @@ public void handleMessage(Object msg) {
                 }
                 if (msg instanceof LastHttpContent) {
                     responseBuilder.body(baos.toByteArray());
+                    try {
+                        baos.close();
```
Comment (patriot1burke) [inline extensions/azure-functions-http/runtime/src/main/java/io/quarkus/azure/functions/resteasy/runtime/BaseFunction.java:162]: Look at the implementation for BAOS.close().  its empty.

File: extensions/amazon-lambda-http/runtime/src/main/java/io/quarkus/amazon/lambda/http/LambdaHttpHandler.java [hunk 3]
```diff
@@ -122,7 +125,12 @@ public void handleMessage(Object msg) {
                             responseBuilder.setBase64Encoded(true);
                             responseBuilder.setBody(Base64.getMimeEncoder().encodeToString(baos.toByteArray()));
                         } else {
-                            responseBuilder.setBody(new String(baos.toByteArray(), "UTF-8"));
+                            responseBuilder.setBody(new String(baos.toByteArray(), StandardCharsets.UTF_8));
+                        }
+                        try {
+                            baos.close();
```
Comment (patriot1burke) [inline extensions/amazon-lambda-http/runtime/src/main/java/io/quarkus/amazon/lambda/http/LambdaHttpHandler.java:131]: Again, this just makes the code less readable.  BAOS.close() doesn't do anything.

File: extensions/azure-functions-http/runtime/src/main/java/io/quarkus/azure/functions/resteasy/runtime/BaseFunction.java (hunk 2 above)
Comment (loicmathieu) [reply extensions/azure-functions-http/runtime/src/main/java/io/quarkus/azure/functions/resteasy/runtime/BaseFunction.java:162]: Oh yes !
So I'd rather remove the closing of the BAOS from google-cloud-functions-http

File: extensions/amazon-lambda-http/runtime/src/main/java/io/quarkus/amazon/lambda/http/LambdaHttpHandler.java (hunk 1 above)
Comment (loicmathieu) [reply extensions/amazon-lambda-http/runtime/src/main/java/io/quarkus/amazon/lambda/http/LambdaHttpHandler.java:42]: Sorry, this is standard constant naming style. I assume it didn't hurts to change them.

Comment (loicmathieu) [reply extensions/amazon-lambda-http/runtime/src/main/java/io/quarkus/amazon/lambda/http/LambdaHttpHandler.java]: Reverted

File: extensions/amazon-lambda-http/runtime/src/main/java/io/quarkus/amazon/lambda/http/LambdaHttpHandler.java (hunk 3 above)
Comment (loicmathieu) [reply extensions/amazon-lambda-http/runtime/src/main/java/io/quarkus/amazon/lambda/http/LambdaHttpHandler.java]: Reverted

**Comment by gsmet** [comment]
@patriot1burke are you good with it now?

**Comment by loicmathieu** [comment]
@patriot1burke can we merge this one to align the buffer size for AWS, Azure and Google ?
The choosen size is the default used inside the JVM so it should be a sane default.

**Review by gsmet — APPROVED** [review]

**Pull request merged**